.github/workflows/ci.yml

```diff
@@ -68,26 +91,34 @@ jobs:
         name: Unshallow
         run: git fetch --prune --unshallow
       -
-        name: Read go version (Unix)
-        if: ${{ runner.os != 'Windows' }}
-        id: go-version-unix
+        name: Read go version
+        id: go-version
+        shell: bash
         run: |
           content=`cat ./.go-version`
           echo "::set-output name=content::$content"
-      -
-        name: Read go version (Windows)
-        if: ${{ runner.os == 'Windows' }}
-        id: go-version-win
-        run: |
-          $content = Get-Content .\.go-version -Raw
-          echo "::set-output name=content::$content"
       -
         name: Set up Go
         uses: actions/setup-go@v2
         with:
           # TODO: Replace with go-version-from-file when it is supported
           # https://github.com/actions/setup-go/pull/62
-          go-version: ${{ steps.go-version-unix.outputs.content || steps.go-version-win.outputs.content }}
+          go-version: ${{ steps.go-version.outputs.content }}
+      - uses: actions/cache@v2
+        with:
+          # In order:
+          # * Module download cache
+          # * Build cache (Linux)
+          # * Build cache (Mac)
+          # * Build cache (Windows)
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/go-build
+            ~/Library/Caches/go-build
+            ~/AppData/Local/go-build
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
       -
```

Comment:
```suggestion
          key: ${{ runner.os }}-${{ runner.arch }}-go-${{ hashFiles('**/go.sum') }}
          restore-keys: |
            ${{ runner.os }}-${{ runner.arch }}-go-
      -
```
Nitpick - I believe the cache _could_ differ by architecture as well. Admittedly we don't run anywhere else other than on Intel right now, but this minor change would allow us to do in the future.

Reply:
We have a [10GB max limit](https://github.com/actions/cache#cache-limits) of the size of things we can cache, so we'll have to monitor it. My local `go-build` is 1GB, not including the other locations, so multiplying that by the amount of platforms/architectures and we could hit that limit soon.

Reply:
Yep - I don't think that would affect the existing workflow though - because `runner.arch` will be `amd64` in all cases - so the same amount of data would just get cached under a different key.

Again - not a big deal - not a blocker for the PR!